// @dwp/dwp-frontend/components/side-navigation/side-navigation.js
function SideNavigation($module) {
  this.$module = document.querySelector($module);
  this.$sideNavLinks = document.querySelector(
    "[data-module='" + this.$module.dataset.target + "']",
  );
  this.$buttonArrow = document.querySelector(".dwp-side-navigation__chevron");
  this.$buttonLabel = document.querySelector(
    ".dwp-side-navigation__section-toggle-text",
  );
}

SideNavigation.prototype.init = function init() {
  this.$module.addEventListener(
    "click",
    this.eventHandlers.toggleSideNavigation.bind(this),
  );
};

SideNavigation.prototype.eventHandlers = {
  toggleSideNavigation(event) {
    event.preventDefault();
    const expanded =
      this.$module.getAttribute("aria-expanded") === "false" ? true : false;
    const labelText = expanded
      ? this.$module.dataset.labelHide
      : this.$module.dataset.labelShow;

    this.$module.setAttribute("aria-expanded", expanded);
    this.$sideNavLinks.classList.toggle("dwp-side-navigation--mobile");
    this.$buttonLabel.innerHTML = labelText;
    this.$buttonArrow.classList.toggle("dwp-side-navigation__chevron--down");
  },
};

module.exports = { SideNavigation };
